crypto/bio/bio_print.c

```diff
@@ -409,19 +440,22 @@ fmtstr(char **sbuffer,
     if (flags & DP_F_MINUS)
         padlen = -padlen;
 
-    while ((padlen > 0) && (max < 0 || cnt < max)) {
+    while ((padlen > 0) && (max < 0 || cnt < max)
+           && !is_eob(buffer, currlen, maxlen)) {
```

Comment:
perhaps I'm missing something here. what is the purpose of newly introduced `is_eob()` check? do I get it right it stands for 'is it end-of-buffer'?

Is it intentional to place it here? and not into `doapr_outch()` function? I'm just asking as I'd like to be sure if it is OK to let `fmtint()` function to return 1 in case `is_eob()` kicks in.

currently any error which happens in `doapr_outch()` makes `fmtint()` to return 0. the new behavior introduced by `is_eob()` is what bothers as I'm currently unable to tell if it is desired or not.